### PR TITLE
Date addition

### PR DESCRIPTION
**Pull request by loganwright**
taking over #32 and #33

### CODE REVIEW

**Comment by loganwright** [comment]
There's still a lot of questions about how we should serialize the date.

Currently, I think incoming support is solid, we're pretty flexible w/ Dates, Strings, and Timestamps in a really extensible way.

Outgoing support is more difficult, since a Node can be represented as a Date, it becomes the serialization's job to know how it should be serialized, for example ... JSON should know whether the Date should be represented as Timestamp or String, currently there's no way to indicate this ... we could default 8601 which is (as standard as possible) for JSON, and say other usecases require users to implement manually.

@BrettRToomey @tannernelson

File: Sources/Node/Convertible/Date+Convertible.swift [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        switch node {
+        case let .date(date):
+            self = date
+        case let .number(number):
+            self = try Date.incomingTimestamp(number)
+        case let .string(string):
+            guard let date = Date.incomingDateFormatters.lazy.flatMap({ $0.date(from: string) }).first else { fallthrough }
+            self = date
+        default:
+            throw NodeError(node: node, expectation: "\(Date.self), formatted time string, or timestamp", key: nil)
```
Comment (tanner0101) [inline Sources/Node/Convertible/Date+Convertible.swift:74]: Can we make a convenience init for `NodeError` with `key: String? = nil` so we don't need `key: nil` everywhere?

**Review by tanner0101 — APPROVED** [review]
This looks really nice.

File: Sources/Node/Convertible/Date+Convertible.swift (hunk 1 above)
Comment (loganwright) [reply Sources/Node/Convertible/Date+Convertible.swift:74]: Yea, definitely. Came from enum and overlooked the default arg!

**Pull request merged**